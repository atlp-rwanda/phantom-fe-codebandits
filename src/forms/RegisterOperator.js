import React from 'react';
import { toast } from 'react-toastify';
import { FormComponent } from './FormComponent.js';
import { operatorInputs } from './FormInputs.js';

export const LabelComponent = ({ name, htmlFor }) => {
  return (
    <label
      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
      htmlFor={htmlFor}
    >
      {name}
    </label>
  );
};
export const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

function RegisterOperator() {
  const handleReg = async (data) => {
    return await sleep(3000).then((response) => {
      console.log('Data sent');
      toast(`A new operator with name ${data.firstname}`);
    });
  };
  return (
    <FormComponent
      inputs={operatorInputs}
      type={'Register Operator'}
      callback={handleReg}
      redirect={'/dashboard/operator'}
    ></FormComponent>
  );
}

export default RegisterOperator;