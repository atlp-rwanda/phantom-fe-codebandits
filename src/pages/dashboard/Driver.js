import React from 'react';

function Driver() {
  return <h1 className="text-black">Drivers</h1>;
}

export default Driver;
