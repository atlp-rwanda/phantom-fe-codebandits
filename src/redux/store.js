import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './reducers/authReducer.js';
import { counterReducer } from './reducers/counterReducer.js';
import { updateProfileReducer } from './reducers/updateProfileReducer.js';
import { profileReducer } from './reducers/profileReducer.js';
import { managementReducer } from './reducers/managementReducer.js';
import { dashboardReducer } from './reducers/dashboardReducer.js';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    profileUpdates: updateProfileReducer,
    profileInfo: profileReducer,
    auth: authReducer,
    management: managementReducer,
    dashboard: dashboardReducer
  }
});
