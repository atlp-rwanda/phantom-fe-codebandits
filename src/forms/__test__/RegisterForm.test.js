import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { store } from '../../redux/store.js';
import Driver from '../../pages/dashboard/Driver.js';
import Operator from '../../pages/dashboard/Operator.js';
import RegisterOperator, {
  LabelComponent
} from '../RegisterOperator.js';

describe('Registration', () => {
  describe('Operators', () => {
    it('Should render the operator form', () => {
      const elem = renderer
        .create(
          <BrowserRouter>
            <Operator />
          </BrowserRouter>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
    it('Should render the driver form', () => {
      const elem = renderer
        .create(
          <Provider store={store}>
            <BrowserRouter>
              <Driver />
            </BrowserRouter>
          </Provider>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
  });

  describe('Component', () => {
    const wrapper = shallow(
      <LabelComponent htmlFor={'test'} name={'test'}></LabelComponent>
    );
    it('Should return test', () => {
      expect(wrapper.text()).toEqual('test');
    });
  });
});
