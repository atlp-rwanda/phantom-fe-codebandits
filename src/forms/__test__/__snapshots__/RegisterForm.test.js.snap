// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Should render the driver form 1`] = `
<h1
  className="text-black"
>
  Drivers
</h1>
`;

exports[`Registration Should render the operator form 1`] = `
<div
  className="flex flex-row h-full w-full items-center justify-center"
>
  <h1
    className="text-3xl font-bold"
  >
    The operators page
  </h1>
</div>
`;
